designs/static-capacity.md

```diff
@@ -0,0 +1,141 @@
+# Karpenter - Static Capacity
+
+## Background
+Karpenter currently operates as a dynamic cluster autoscaler, automatically adjusting node counts based on pending pod demand. However, several important use cases require maintaining a fixed set of nodes:
+
+1. Performance-critical applications where just-in-time provisioning latency is unacceptable
+2. Workloads that need guaranteed capacity regardless of pod demand  
+3. Scenarios where fixed node counts are preferred
+
+Currently, users attempt to achieve static capacity through workarounds:
+- Running placeholder pods to maintain minimum capacity
+- Using separate node management tools alongside Karpenter (MNG/Fargate/Self-managed)
+- Configuring complex provisioner requirements to approximate static behavior
+
+## Proposal
+Extend the existing NodePool resource to support static provisioning capabilities by adding new fields:
+
+```yaml
+spec:
+  # Makes NodePool static when specified
+  replicas: 5  # Number of nodes to maintain
+```
+
+Key aspects:
+1. Static NodePools maintain fixed node count regardless of pod demand
+2. Will not be considered as consolidation candidate (at least in v1alpha1)
+3. Nodes are spread across AZs by default
+4. Inherit existing Karpenter features like drift detection and interruption handling
```

Comment:
Do you mean spot interruption handling? You should drop this part, since this is really a cloud provider feature